Tests/Microsoft.AppCenter.Test.Windows/Ingestion/Http/RetryableIngestionTest.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AppCenter.Ingestion;
+using Microsoft.AppCenter.Ingestion.Http;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Microsoft.AppCenter.Test.Windows.Ingestion.Http
+{
+    [TestClass]
+    public class RetryableIngestionTest : BaseIngestionTest
+    {
+        private static readonly TimeSpan[] Intervals =
+        {
+            TimeSpan.FromSeconds(1),
+            TimeSpan.FromSeconds(1),
+            TimeSpan.FromSeconds(1) 
+        };
+        private IIngestion _retryableIngestion;
+
+        [TestInitialize]
+        public void InitializeRetryableTest()
+        {
+            _adapter = new Mock<IHttpNetworkAdapter>();
+            _retryableIngestion = new RetryableIngestion(new HttpIngestion(_adapter.Object), Intervals);
+        }
+
+        /// <summary>
+        /// Verify behaviour without exceptions.
+        /// </summary>
+        [TestMethod]
+        public async Task RetryableIngestionSuccess()
+        {
+            SetupAdapterSendResponse(HttpStatusCode.OK);
+            using (var call = _retryableIngestion.Call(AppSecret, InstallId, Logs))
+            {
+                await call.ToTask();
+                VerifyAdapterSend(Times.Once());
+            }
+
+            // No throw any exception
+        }
+
+        /// <summary>
+        /// Verify that retrying on recoverable exceptions.
+        /// </summary>
+        [TestMethod]
+        public async Task RetryableIngestionRepeat1()
+        {
+            // RequestTimeout - retryable
+            SetupAdapterSendResponse(HttpStatusCode.RequestTimeout, HttpStatusCode.OK);
+            var start = DateTime.Now;
+            using (var call = _retryableIngestion.Call(AppSecret, InstallId, Logs))
+            {
+                await Task.Delay(start.AddSeconds(0.5) - DateTime.Now);
+                Assert.IsFalse(call.IsCompleted);
+                VerifyAdapterSend(Times.Exactly(1));
+
+                await Task.Delay(start.AddSeconds(1.5) - DateTime.Now);
```

Comment:
Since `Timer` is used instead of async tasks which we changed to completed one in tests, real delays are used here. It even made the tests more readable